src/mesh/Data.cpp

```diff
@@ -21,13 +23,15 @@ Data::Data()
 Data::Data(
     std::string name,
     DataID      id,
-    int         dimensions)
+    int         dimensions,
+    int         interpolationOrder)
     : _values(),
       _name(std::move(name)),
       _id(id),
       _dimensions(dimensions)
 {
   PRECICE_ASSERT(dimensions > 0, dimensions);
+  _ptrWaveform = time::PtrWaveform(new time::Waveform(interpolationOrder));
```

Comment:
I would suggest to only move the creation of the waveform as soon as we need it (not earlier). Moving it is not a problem and right now it makes the configuration easier, if the configuration does not have to worry about the waveform, since data is taking care of this.

Reply:
Sure, we can move it later, but this is in a way inconsistent with how we create other classes (mapping, acceleration, ...). If possible, they are normally created in the configuration.